### DISCUSSION TITLE
Allow Notes to be Exported as well as Node Titles

### DISCUSSION THREAD

**Discussion by Foxxsayy** [freeplane/freeplane#1336] [Ideas]
Please correct me if I've missed an option somewhere, but I have just tried most of the export options and only the Java applet option exports notes as far as I can tell.  I've been searching for a solution to help me draft papers and such, and I like how FreePlane allows so much text in the actual nodes. However, if you can't export notes along with the node text, you lose a huge amount of information or have to do a ton of work with a workaround like copy-pasting the Java applet into word and then formatting it all again.

**It is possible to get an option allows notes to be exported along with the rest of the map - perhaps indented under the topic? Especially to word files?** 

(The export option to Word is otherwise pretty good, aside from some minor features like no indentation options.)

**Comment by macmarrum** [comment]
I rarely use the built-in export functionality. Instead, I add my own export formats via Groovy scripting.

Here's my library that can export to Markdown and to CSV.
Markdown covers core, details and note by default.
CSV covers a single node part, i.e. either core (the default) or details or note.

You can download the library [Export.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/lib/Export.groovy) and save it in `<user-dir>/lib` (you can do `Tools->Open user directory`)

You can also download (and save in `<user-dir>/scripts`) scripts that use the library to do the actual export. Each script exports a branch starting from a selected node.
- [exportToMarkdownFile.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/exportToMarkdownFile.groovy)
- [exportToCsvFile.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/exportToCsvFile.groovy)

Please keep in mind to restart Freeplane so that the new scripts appear.

You'll find the scripts in the menu `Mac1->Export`.

Please feel free to remove `"/menu_bar/Mac1/Export"` in the first line of each script to have them appear in `Tools->Scripts`.

You can change the settings of each export by editing the script files (e.g. in [Notepad++](https://notepad-plus-plus.org/)).

E.g. to export notes to CSV, you can change the following line in [exportToCsvFile.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/exportToCsvFile.groovy) by adding the settings `[nodePart: 'NOTE']`.
```groovy
Export.toCsvFile(file, node, [nodePart: 'NOTE'])
```

---

Here are the default settings
```groovy
def mdSettings = [h1: 'NODE', details: 'HLB', note: 'PLAIN']
def csvSettings = [sep: ',', eol: '\n', newlineReplacement: '\r', nodePart: 'CORE']
```

and possible values of some of the settings
```groovy
enum NodePart { CORE, DETAILS, NOTE }
```

```groovy
enum MdInclude { NONE, QUOTE, HLB, QUOTE_HLB, CODE, PLAIN }
```
<p>Markdown Inclusion</p>
<p>Reflects how input should be treated in Markdown</p>
<ul>
    <li>NONE - not included</li>
    <li>QUOTE - as a quote; lines will be merged</li>
    <li>HLB - hard line breaks for each line</li>
    <li>QUOTE_HLB - as a quote with a hard line break for each line</li>
    <li>CODE - as a code block</li>
    <li>PLAIN - plain text, without any modification</li>
</ul>

```groovy
enum MdH1 { NONE, ROOT, NODE }
```
<p>Markdown Heading1</p>
<p>Reflects where headings are counted from when Level Styles are encountered</p>
<ul>
<li>ROOT - root: #, Level 1: ##, etc.</li>
<li>NODE - the first encountered Level Style: #, the second: ##, etc</li>
</ul
